src/Microsoft.Data/PrimitiveDataFrameColumn.cs

```diff
@@ -652,5 +652,24 @@ protected internal override Delegate GetDataViewGetter(DataViewRowCursor cursor)
 
         private static ValueGetter<double> CreateDecimalValueGetterDelegate(DataViewRowCursor cursor, PrimitiveDataFrameColumn<decimal> column) =>
             (ref double value) => value = (double?)column[cursor.Position] ?? double.NaN;
+
+        protected internal override void AddValueUsingCursor(DataViewRowCursor cursor, DataViewSchema.Column column)
+        {
+            long row = cursor.Position;
+            T value = default;
+            cursor.GetGetter<T>(column)(ref value);
```

Comment:
The `Getter` delegate should be cached outside of the "loop" of cursoring over an IDataView. Getting this for each "cell" in the IDataView is going to be super slow.

Check out [https://github.com/dotnet/machinelearning/blob/master/docs/code/IDataViewImplementation.md#why-getgetter](https://github.com/dotnet/machinelearning/blob/master/docs/code/IDataViewImplementation.md#why-getgetter)